### PR TITLE
Js/fix odometry

### PR DESCRIPTION
**Pull request by Roboterbastler**

### CODE REVIEW

File: include/tricycle_drive_controller/odometry.h [hunk 1]
```diff
@@ -152,8 +152,10 @@ class Odometry
      * \brief Sets the wheel parameters: radius and separation
      * \param wheel_base   whell base [m]
      * \param wheel_radius     Wheel radius [m]
+     * \param front_wheel_caster_offset Offset of the front wheel center from the
+     * caster joint axis [m]
      */
-    void setWheelParams(double wheel_base, double wheel_radius);
+    void setWheelParams(double wheel_base, double wheel_radius, double front_wheel_caster_offset);
```
Comment (bochen87) [inline include/tricycle_drive_controller/odometry.h]: maybe call it "front_wheel_position_offset" or "front_wheel_offset_to_center" ?

File: include/tricycle_drive_controller/tricycle_drive_controller.h [hunk 2]
```diff
@@ -137,6 +137,12 @@ class TricycleDriveController : public controller_interface::MultiInterfaceContr
     /// Wheel radius (assuming it's the same for the left and right wheels):
     double wheel_radius_;
 
+    /// Offset of the front wheel center from the caster joint axis [m]
+    double front_wheel_caster_offset_;
+
+    /// Angular offset of the steering angle
+    double front_wheel_steering_offset_;
```
Comment (bochen87) [inline include/tricycle_drive_controller/tricycle_drive_controller.h]: "front_wheel_angular_offset" or "front_wheel_steering_angle_offset" ?

File: src/tricycle_drive_controller.cpp [hunk 3]
```diff
@@ -264,7 +268,7 @@ TricycleDriveController::update(const ros::Time& time, const ros::Duration& peri
 {
     // COMPUTE AND PUBLISH ODOMETRY
     if (real_hw_) {
-        odometry_.updateFromHWEncoders(front_wheel_cmd.front().getVelocity(), front_wheel_caster_cmd.front().getPosition(), time);
+        odometry_.updateFromHWEncoders(front_wheel_cmd.front().getVelocity(), front_wheel_caster_cmd.front().getPosition() + front_wheel_steering_offset_, time);
```
Comment (bochen87) [inline src/tricycle_drive_controller.cpp]: maybe put the front_wheel_steering_offset variable also in the odometry.cpp? That way all the offset stuff is in one place if you're searching for it.

**Review by bochen87 — APPROVED** [review]
maybe put the front_wheel_steering_offset variable also in the odometry.cpp? That way all the offset stuff is in one place if you're searching for it.

File: src/tricycle_drive_controller.cpp (hunk 3 above)
Comment (Roboterbastler) [reply src/tricycle_drive_controller.cpp]: Well, I thought the offset isn't really related to the odometry calculation but just a modification of the values sent to and read from the hardware. I would say it's more consistent to keep it out of the odometry and also use the parameter in only one file...

File: include/tricycle_drive_controller/tricycle_drive_controller.h (hunk 2 above)
Comment (Roboterbastler) [reply include/tricycle_drive_controller/tricycle_drive_controller.h]: Yeah, maybe that makes it more clear

**Comment by Roboterbastler** [comment]
The renamings are done now

**Comment by bochen87** [comment]
looks good!

**Comment by Roboterbastler** [comment]
[Rework Tricycle Driver](https://trello.com/c/kE8ex12W/362-rework-tricycle-driver)

**Pull request merged**